packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/edit-rules-form/edit-rules-form.tsx

```diff
@@ -39,65 +37,38 @@ const EditRules = zod.object({
           .min(1, { message: i18n.t("promotions.form.required") }),
       ]),
       required: zod.boolean().optional(),
+      disguised: zod.boolean().optional(),
       field_type: zod.string().optional(),
     })
   ),
 })
 
 export const EditRulesForm = ({
   promotion,
-  rules,
   ruleType,
-  attributes,
-  operators,
   handleSubmit,
   isSubmitting,
 }: EditPromotionFormProps) => {
   const { t } = useTranslation()
-  const requiredAttributes = attributes?.filter((ra) => ra.required) || []
-  const requiredAttributeValues = requiredAttributes?.map((ra) => ra.value)
-  const disguisedRules =
-    getDisguisedRules(promotion, requiredAttributes, ruleType) || []
   const [rulesToRemove, setRulesToRemove] = useState([])
 
   const form = useForm<zod.infer<typeof EditRules>>({
-    defaultValues: {
-      rules: [...disguisedRules, ...rules].map((rule) => ({
-        id: rule.id,
-        required: requiredAttributeValues.includes(rule.attribute),
-        attribute: rule.attribute!,
-        operator: rule.operator!,
-        values: Array.isArray(rule?.values)
-          ? rule?.values?.map((v: any) => v.value!)
-          : rule.values!,
-      })),
-    },
+    defaultValues: { rules: [], type: promotion.type },
     resolver: zodResolver(EditRules),
   })
 
-  const { fields, append, remove, update } = useFieldArray({
-    control: form.control,
-    name: "rules",
-    keyName: "rules_id",
-  })
-
   const handleFormSubmit = form.handleSubmit(handleSubmit(rulesToRemove))
 
   return (
     <RouteDrawer.Form form={form}>
       <form onSubmit={handleFormSubmit} className="flex h-full flex-col">
         <RouteDrawer.Body>
           <RulesFormField
-            form={form}
+            form={form as any}
```

Comment:
What's the issue with the typings here, maybe we can resolve it without any?

Reply:
This was legacy, forgot to remove it. 🤞🏻 Good catch